Common/Data/Market/IBar.cs

```diff
@@ -0,0 +1,45 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+
+namespace QuantConnect.Data.Market
+{
+    /// <summary>
+    /// Base Bar Class: Open, High, Low and Close.
+    /// </summary>
+    public interface IBar
+    {
+        /// <summary>
+        /// Opening price of the bar: Defined as the price at the start of the time period.
+        /// </summary>
+        decimal Open { get; set; }
+
+        /// <summary>
+        /// High price of the bar during the time period.
+        /// </summary>
+        decimal High { get; set; }
+
+        /// <summary>
+        /// Low price of the bar during the time period.
+        /// </summary>
+        decimal Low { get; set; }
+
+        /// <summary>
+        /// Closing price of the bar. Defined as the price at Start Time + TimeSpan.
+        /// </summary>
+        decimal Close { get; set; }
```

Comment:
We have setters on OHLC at TradeBar, so I didn't gave ant thoughts on that requirement.